AirCasting/Notes/NotesHandler.swift

```diff
@@ -60,18 +60,22 @@ class NotesHandlerDefault: NSObject, NotesHandler, NSFetchedResultsControllerDel
             }
         }
     }
-    
+
     func deleteNote(note: Note, completion: @escaping () -> Void) {
-        measurementStreamStorage.accessStorage { [self] storage in
+        measurementStreamStorage.accessStorage { [weak self] storage in
+            guard let self = self else { return }
             do {
-                try storage.deleteNote(note, for: sessionUUID)
-                fetchSession(storage: storage) { session in
+                try storage.deleteNote(note, for: self.sessionUUID)
+                self.fetchSession(storage: storage) { session in
                     self.sessionUpdateService.updateSession(session: session) { result in
                         switch result {
                         case .success(let updateData):
-                            try? storage.updateVersion(for: self.sessionUUID, to: updateData.version)
-                            Log.info("Notes successfully updated")
-                            completion()
+                            self.measurementStreamStorage.accessStorage { [weak self] storage in
+                                guard let self = self else { return }
```

Comment:
hmm, using `weak self` in this `deleteNote` might not be the best. I don't see any retain cycles possible, and by using `weak` we don't hold on to the `self` till delete is finished. So this kind of code:
```Swift
func asd() {
    let handler = NotesHandlerDefault(...)
    handler.deleteNote(note: note) {
        print("Deleted!")
    }
}
```

will never execute `print("Deleted!")`.

IDK if that's realistic here